apiserver/facades/client/modelmanager/modelmanager.go

```diff
@@ -1347,11 +1346,8 @@ func (m *ModelManagerAPI) ModifyModelAccess(ctx context.Context, args params.Mod
 				},
 				Access: modelAccess,
 			},
-			AddUser:  true,
-			External: &external,
-			ApiUser:  user.NameFromTag(m.apiUser),
-			Change:   permission.AccessChange(arg.Action),
-			Subject:  user.NameFromTag(targetUserTag),
+			Change:  permission.AccessChange(arg.Action),
+			Subject: user.NameFromTag(targetUserTag),
```

Comment:
issue: This changes the expected behavior of the facade here. If not adding a user in the permission domain update user, you must handle the case where a user doesn't exist when ModifyModelAccess is called and add the user.

Reply:
I'm changing it here to match the behaviour of 3.6. If the user does not exist locally then they are added, otherwise not. This is what I have implemented in this PR.

See the [state code here with this behaviour](https://github.com/Aflynn50/juju/blob/5e07f2c9d041d674a524aaff0a7a599023b40ae1/state/clouduser.go#L27).

Also this QA
```
➜  ~ ctrl36 juju grant-cloud asda admin lxd
ERROR could not grant cloud access: user "asda" does not exist locally: user "asda" not found
➜  ~ ctrl36 juju grant-cloud jim@external admin lxd
➜  ~ ctrl36 
```

Reply:
By fixing the behavior of external, you're also breaking the 3.6 behavior for non external users.

Reply:
The method above calls through to here:
https://github.com/Aflynn50/juju/blob/5e07f2c9d041d674a524aaff0a7a599023b40ae1/state/useraccess.go#L43-L51
This then calls to here:
https://github.com/Aflynn50/juju/blob/5e07f2c9d041d674a524aaff0a7a599023b40ae1/state/useraccess.go#L62-L72

It is adding them as a model user, not as a controller user. For a model user to be added, if they are local they already need to exist as a local user on the controller. This requirement does not exist for external users.

We are getting rid of model users as part of this work, all that is left of the model user collection is the model user permissions, which are now stored on the controller.

If there is a scenario that I've missed that works in 3.x and not in this patch I will fix it, but I do not believe this is a behaviour change from 3.x.